www-plugins/browserpass/browserpass-2.0.13.ebuild

```diff
@@ -0,0 +1,49 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+EGO_PN=github.com/dannyvankooten/browserpass
+
+if [[ ${PV} == 9999 ]]; then
+	inherit golang-vcs
+else
+	KEYWORDS="~amd64"
+	SRC_URI="https://${EGO_PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+	inherit golang-vcs-snapshot
+fi
+inherit golang-build
+
+DESCRIPTION="WebExtension host binary for pass, a UNIX password manager"
+HOMEPAGE="https://github.com/dannyvankooten/browserpass"
+LICENSE="MIT"
+SLOT="0"
+RDEPEND="app-crypt/gnupg"
+DEPEND="${RDEPEND}
+	dev-go/fuzzy:=
+	dev-go/twofactor:=
+	dev-go/zglob:="
+
+DOCS=( CONTRIBUTING.md README.md )
+
+src_compile() {
+	EGO_PN="${EGO_PN}/cmd/browserpass" golang-build_src_compile
```

Comment:
I'm really lost here. Is the global value of EGO_PN really correct if you need to redefine it here?

Reply:
`golang-build.eclass` is a bit weird in that it uses `EGO_PN` to create the path in `PORTAGE_TMPDIR`,  and ALSO uses that path in order to decide where to start `go build`, which is kind of annoying.
e.g. `${PORTAGE_TMPDIR}/work/${P}/src/${EGO_PN}`.

So in order to build the browserpass commandline application I need to add "/cmd/browserpass" to `EGO_PN` during the compile phase (the subdirectory in which `main.go` is located) in order for go to compile it -- otherwise it will only create a browserpass library, instead of a binary.

Personally I think the eclass should be fixed, but atm this seemed like an easier to maintain solution than copying the body of golang-build_src_compile over from the eclass and adjusting the `EGO_PN` line in there.